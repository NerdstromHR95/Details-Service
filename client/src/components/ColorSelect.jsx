import React from 'react';
import ColorSelectMenu from './ColorSelectMenu.jsx'


class ColorSelect extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			options: this.props.colors,
			selected: this.props.selected,
			menuOpen: this.props.menuDown
		}
	}

	componentDidUpdate(prevProps) {
		if (prevProps.selected !== this.props.selected) {
			this.setState({
				selected: this.props.selected
			})
		}
		if (this.state.menuOpen !== this.props.menuDown) {
			this.setState({
				menuOpen: this.props.menuDown
			})
		}
	}

	render() {
		const menuOpen = this.state.menuOpen
		if (menuOpen) {
			return <ColorSelectMenu toggleMenu={this.props.toggleMenu} changeColor={this.props.changeColor} options={this.state.options}/>
		} else {
			return (
			  <div>
				<div className='selectedColor' onClick={this.props.toggleMenu}>
				  {this.state.selected === 'Color' ? <span className='smallIndent menuDefault'>{this.state.selected}</span> : <span className='smallIndent'>{this.state.selected}</span>}
				  <i className="fas fa-angle-down arrow"></i>
				</div>
			  </div>
			)
		}
	}
}

export default ColorSelect;