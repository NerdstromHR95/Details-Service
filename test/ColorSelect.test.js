import React from 'react';
import ColorSelect from '../client/src/components/ColorSelect.jsx';
import renderer from 'react-test-renderer';

test('Should render closed menu', () => {
	const component = renderer.create(<ColorSelect menuDown={false} colors={['Black', 'White', 'Blue', 'Red', 'Green']} />);
	let tree = component.toJSON();

	expect(tree).toMatchSnapshot();
})

test('Should render open menu', () => {
	const component = renderer.create(<ColorSelect menuDown={true} colors={['Black', 'White', 'Blue', 'Red', 'Green']} />);
	let tree = component.toJSON();

	expect(tree).toMatchSnapshot();
})

test('Should render Red as selected', () => {
	const component = renderer.create(<ColorSelect menuDown={false} selected={'Red'} />);
	let tree = component.toJSON();

	expect(tree).toMatchSnapshot();	
})