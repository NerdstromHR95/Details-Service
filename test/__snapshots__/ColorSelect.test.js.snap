// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should open menu when clicked 1`] = `
<div
  onClick={[Function]}
>
  <div
    className="selectedColor"
  >
    <span
      className="smallIndent"
    />
    <i
      className="fas fa-angle-down arrow"
    />
  </div>
  <div
    className="colorSwatches"
  >
    <svg
      className="circle"
      height="35"
      width="35"
    >
      <circle
        className="frontCircle"
        cx="15"
        cy="15"
        fill="Black"
        id="Black"
        onClick={[Function]}
        r="13"
        stroke="grey"
        strokeWidth="1"
      />
    </svg>
    <svg
      className="circle"
      height="35"
      width="35"
    >
      <circle
        className="frontCircle"
        cx="15"
        cy="15"
        fill="White"
        id="White"
        onClick={[Function]}
        r="13"
        stroke="grey"
        strokeWidth="1"
      />
    </svg>
    <svg
      className="circle"
      height="35"
      width="35"
    >
      <circle
        className="frontCircle"
        cx="15"
        cy="15"
        fill="Blue"
        id="Blue"
        onClick={[Function]}
        r="13"
        stroke="grey"
        strokeWidth="1"
      />
    </svg>
    <svg
      className="circle"
      height="35"
      width="35"
    >
      <circle
        className="frontCircle"
        cx="15"
        cy="15"
        fill="Red"
        id="Red"
        onClick={[Function]}
        r="13"
        stroke="grey"
        strokeWidth="1"
      />
    </svg>
    <svg
      className="circle"
      height="35"
      width="35"
    >
      <circle
        className="frontCircle"
        cx="15"
        cy="15"
        fill="Green"
        id="Green"
        onClick={[Function]}
        r="13"
        stroke="grey"
        strokeWidth="1"
      />
    </svg>
  </div>
</div>
`;

exports[`Should open menu when clicked 2`] = `
<div
  className="menu"
>
  <div
    className="sizeOption"
  >
    <strong
      className="smallIndent"
    >
      Please Choose A Color
    </strong>
  </div>
  <div
    className="colorOption"
    id="Black"
    onClick={[Function]}
  >
    <svg
      className="circle"
      height="30"
      width="30"
    >
      <circle
        cx="15"
        cy="15"
        fill="Black"
        r="13"
        stroke="black"
        strokeWidth="2"
      />
    </svg>
    <span
      className="colorOptionText"
    >
      Black
    </span>
  </div>
  <div
    className="colorOption"
    id="White"
    onClick={[Function]}
  >
    <svg
      className="circle"
      height="30"
      width="30"
    >
      <circle
        cx="15"
        cy="15"
        fill="White"
        r="13"
        stroke="black"
        strokeWidth="2"
      />
    </svg>
    <span
      className="colorOptionText"
    >
      White
    </span>
  </div>
  <div
    className="colorOption"
    id="Blue"
    onClick={[Function]}
  >
    <svg
      className="circle"
      height="30"
      width="30"
    >
      <circle
        cx="15"
        cy="15"
        fill="Blue"
        r="13"
        stroke="black"
        strokeWidth="2"
      />
    </svg>
    <span
      className="colorOptionText"
    >
      Blue
    </span>
  </div>
  <div
    className="colorOption"
    id="Red"
    onClick={[Function]}
  >
    <svg
      className="circle"
      height="30"
      width="30"
    >
      <circle
        cx="15"
        cy="15"
        fill="Red"
        r="13"
        stroke="black"
        strokeWidth="2"
      />
    </svg>
    <span
      className="colorOptionText"
    >
      Red
    </span>
  </div>
  <div
    className="colorOption"
    id="Green"
    onClick={[Function]}
  >
    <svg
      className="circle"
      height="30"
      width="30"
    >
      <circle
        cx="15"
        cy="15"
        fill="Green"
        r="13"
        stroke="black"
        strokeWidth="2"
      />
    </svg>
    <span
      className="colorOptionText"
    >
      Green
    </span>
  </div>
</div>
`;
